crates/typst-library/src/compute/data.rs

```diff
@@ -119,11 +123,44 @@ pub fn csv(
     let Spanned { v: path, span } = path;
     let id = vm.location().join(&path).at(span)?;
     let data = vm.world().file(id).at(span)?;
+    csv_decode(Spanned::new(Readable::Bytes(data), span), delimiter)
+}
 
+/// Reads structured data from a CSV string/bytes.
+///
+/// The CSV string/bytes will be read and parsed into a 2-dimensional array of strings:
+/// Each row in the CSV file will be represented as an array of strings, and all
+/// rows will be collected into a single array. Header rows will not be
+/// stripped.
+///
+/// ## Example { #example }
+/// ```example
+/// #let results = csv.decode(read("data.csv"))
+///
+/// #table(
+///   columns: 2,
+///   [*Condition*], [*Result*],
+///   ..results.flatten(),
+/// )
+/// ```
+///
+/// Display: CSV
+/// Category: data-loading
+#[func]
+pub fn csv_decode(
+    /// CSV data.
+    data: Spanned<Readable>,
+    /// The delimiter that separates columns in the CSV file.
+    /// Must be a single ASCII character.
+    #[named]
+    #[default]
+    delimiter: Delimiter,
+) -> SourceResult<Array> {
+    let Spanned { v: data, span } = data;
+    let data: Bytes = data.into();
```

Comment:
General comment: it would be more efficient to have an `as_slice` method on `Readable` to skip the potential string -> bytes conversion here